### ISSUE TITLE
Licence change: LGPL to Apache-2

### ISSUE THREAD

**Issue by pejobo** [pejobo/beanshell2#110]
The code at http://www.beanshell.org/license.html and the fork at https://code.google.com/archive/p/beanshell2/ have been licensed unter LGPL. Since then the license of the original code has been moved and the license has been changed to Apache-2.0: https://github.com/beanshell/beanshell/blob/master/LICENSE.

I have agreed to make my changes available under an Apache License version 2, to match the licensing change on the original Beanshell project code and to allow re-integration of all changes in Beanshell2 back to the original Beanshell project.

I'll try to get the consent of the original maintainer and other commiters for a dual licence.

Agreement should be expressed here by replying to this issue with "I agree to make my changes available under an Apache License version 2, to match the licensing change on the original Beanshell project code".

**Comment by fschmidt-nabble** [comment]
I agree to make my changes available under an Apache License version 2, to match the licensing change on the original Beanshell project code.

**Comment by mattflaschen** [comment]
I agree to make my changes available under an Apache License version 2, to match the licensing change on the original Beanshell project code.

My email is similar to my GitHub username (matthew.flaschen@, domain omitted here for anti-spam).

**Comment by nickl-** [comment]
Let me start with an apology, I have slipped keeping tabs with the goings on in beanshell.

Good idea @pejobo this is probably the shortest route to happiness. I would've suggested just making beanshell dual licensed for simplicity sake but personally I have no preference or inclination to bike shed over this, my +1 goes towards any resolve.

What I do feel strongly for though is for these two forks to merge as a matter of urgency. When I joined the project last year, I believe I mentioned stacks of enhancements to implement but instead focused my available time on [merge-fork-beanshell2](https://github.com/beanshell/beanshell/tree/merge-fork-beanshell2). Unfortunately real life happened before I was done.

The beanshell2 branch addressed many stability issues and incorporated a lot of needed additions like JSR but at the same time the original fork also progressed with features like static which we don't want to loose. I made extra effort to ensure that we incorporate everything.

I managed to grind my way through every commit from the initial fork up and to commit pejobo/beanshell2@09383fe . I merged everything that was in the master then up to pejobo/beanshell2@4df454b976 and started on the [2.1 branch commits](https://github.com/pejobo/beanshell2/commits/v2.1) already.  There were only 19 commits remoining
 on that page after completing 72 additions to the [merge-fork-beanshell2](https://github.com/beanshell/beanshell/tree/merge-fork-beanshell2) branch.

What we need to do is to finish the merge and release v2.1, archive this branch and continue development at [beanshell/beanshell](/beanshell/beanshell). @pejobo I tried to add you again but you are already a committer at beanshell.org are you aware? I really wish I could spend some time here but I don't even have a java dev environment at the moment. Any ideas?

**Comment by nickl-** [comment]
@pejobo did you read this? Eagerly awaiting for your response....

**Comment by nickl-** [comment]
hmmm... only see LGPL now, is it even allowed to drop gpl? It's not called the [Viral license](https://en.wikipedia.org/wiki/Viral_license) for nothing.

Just popped in to see if you had time to respond. I did some more work on the merge... see post at beanshell org

cio

**Comment by pejobo** [comment]
I'm not a lawyer, therefore I didn't drop LGPL, just asked everyone who contributed to agree to the apache license. And that every new contribution is also dual licensed (not that there are many).

**Comment by nickl-** [comment]
No neither am I... law savvy, just know GPL is different and infects any copyright yet doesn't need to be exclusive. What was beanshell before Apache license?

This is literally the last thing I pay attention to, probably should've done so before I went and merged everything. Now it's committed...well... I guess there's nothing permanent about that can always delete the branch but what a mess. Enough on the license nonsense... good strategy with voicing things and getting agreements. 

@pejobo so glad to see you, hope all is well! You've been absent for a while... on GitHub that is. At least I knew you weren't ignoring me but couldn't help the concerns. Shout if I can help with anything.

Not sure if you've seen my commits and the update post beanshell/beanshell#64 wrote a lot about it already. Done merging everything from beanshell2 now, skipped over a few items we should discuss.

What is your plan now for going forward? Are you cool with being a committer at beanshell org, and taking the work there? There are a few more things that need done before a release, most important is hearing your thoughts and plans. Perhaps we should IM or meet on IRC? Let me know...

**Comment by nickl-** [comment]
## [GPL class path exception](http://www.gnu.org/software/classpath/license.html)

> Linking this library statically or dynamically with other modules is making a combined work based on this library. Thus, the terms and conditions of the GNU General Public License cover the whole combination.
>
> As a special exception, the copyright holders of this library give you permission to link this library with independent modules to produce an executable, regardless of the license terms of these independent modules, and to copy and distribute the resulting executable under terms of your choice, provided that you also meet, for each linked independent module, the terms and conditions of the license of that module. An independent module is a module which is not derived from or based on this library. If you modify this library, you may extend this exception to your version of the library, but you are not obligated to do so. If you do not wish to do so, delete this exception statement from your version.
